src/App.tsx

```diff
@@ -1,14 +1,47 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { getTodos } from './api';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [loading, setLoading] = useState(false);
+  const [showTodos, setShowTodos] = useState('all');
+  const [query, setQuery] = useState('');
+  const [modButton, setModButton] = useState<Todo | null>(null);
+
+  useEffect(() => {
+    setLoading(true);
+    getTodos()
+      .then(items => {
+        setTodos(items);
+      })
+      .finally(() => setLoading(false));
+  }, []);
+
+  const handlerSelect = (e: React.ChangeEvent<HTMLSelectElement>) => {
+    setShowTodos(e.target.value);
+  };
+
+  const handlerInput = (e: React.ChangeEvent<HTMLInputElement> | string) => {
+    if (typeof e === 'string') {
+      setQuery('');
+    } else {
+      setQuery(e.target.value);
+    }
+  };
```

Comment:
The `handlerInput` function accepts a parameter that can be either a string or an event. This dual-purpose parameter might lead to confusion. Consider separating the logic or ensuring the function is only called with one type of parameter.